src/celeritas/ext/VecgeomTrackView.hh

```diff
@@ -365,12 +376,19 @@ CELER_FUNCTION real_type VecgeomTrackView::find_safety(real_type max_radius)
     CELER_EXPECT(!this->is_outside());
     CELER_EXPECT(!this->is_on_boundary());
     CELER_EXPECT(max_radius > 0);
-    real_type safety = detail::BVHNavigator::ComputeSafety(
-        detail::to_vector(this->pos()), vgstate_, max_radius);
+
+    real_type safety = BVHNavigator::ComputeSafety(
+        detail::to_vector(this->pos()),
+        vgstate_
+#if VECGEOM_VERSION < 0x020000
+        ,
+        max_radius
+#endif
+    );
 
     // Since the reported "safety" is negative if we've moved slightly beyond
     // the boundary of a solid without crossing it, we must clamp to zero.
-    return max<real_type>(safety, 0);
+    return clamp(safety, 0., max_radius);
```

Comment:
This should be reverted. It's fine if we return a max radius that's larger. If this changes test results, those should be updated. (Also, you're assuming real_type is double here.)